### PR TITLE
First try for shareWithField error message

### PR DESCRIPTION
**Pull request by Julian1998**
#17408 @PVince81 @MorrisJobke Here my first try - now I need a little help for a nice looking error message

@jancborchardt said that the error message should appear in the dropdown - so I need the response function. But it has no String parameter... so how to use it? :)

### CODE REVIEW

File: core/js/sharedialogview.js [hunk 1]
```diff
@@ -199,6 +199,7 @@
 							$('.shareWithField').autocomplete("option", "autoFocus", true);
 							response(suggestions);
 						} else {
+							OC.Notification.showTemporary(t('core', 'No users or groups found for ' + $('.shareWithField').val()));
```
Comment (PVince81) [inline core/js/sharedialogview.js]: When translating messages with placeholders, you should do it this way:
`t('core', 'Some message about {topic} which is nice', {topic: 'teh topic'})`

**Comment by Julian1998** [comment]
@PVince81 @jancborchardt @MorrisJobke

**Comment by MorrisJobke** [comment]
Looks quite good, nice work. There is only one case left: If you entered a wrong username and then delete everything the message should be gone.

This is how it looks and I like it:

bildschirmfoto 2016-01-22 um 11 23 49

File: core/js/sharedialogview.js [hunk 2]
```diff
@@ -196,9 +196,16 @@
...
 
 						if (suggestions.length > 0) {
+							$('.shareWithField').removeClass('error');
+							$('.shareWithField').tooltip('hide')
 							$('.shareWithField').autocomplete("option", "autoFocus", true);
 							response(suggestions);
+							
 						} else {
+							$('.shareWithField').addClass('error');
+							$('.shareWithField').attr('title', t('core', 'No users or groups found for ' + $('.shareWithField').val()));
```
Comment (DeepDiver1975) [inline core/js/sharedialogview.js]: translations will not work here - no dynamic content can be translated.

```js
t('core', 'No users or groups found for {search}', {search: $('.shareWithField').val()})

File: core/js/sharedialogview.js [hunk 3]
```diff
@@ -196,9 +196,16 @@
 						var suggestions = users.concat(groups).concat(remotes);
 
 						if (suggestions.length > 0) {
+							$('.shareWithField').removeClass('error');
```
Comment (DeepDiver1975) [inline core/js/sharedialogview.js]: please reuse selectors

```js
$('.shareWithField').removeClass('error')
    .tooltip()
    .autocomplete();

File: core/js/sharedialogview.js [hunk 4]
```diff
@@ -196,9 +196,16 @@
...
+							$('.shareWithField').tooltip('hide')
 							$('.shareWithField').autocomplete("option", "autoFocus", true);
 							response(suggestions);
+							
 						} else {
+							$('.shareWithField').addClass('error');
```
Comment (DeepDiver1975) [inline core/js/sharedialogview.js]: same - please reuse selectors

**Comment by jancborchardt** [comment]
Ok nice, I thought it was a yellow error notification. Tooltip is awesome, good work! :+1:

**Comment by Julian1998** [comment]
@MorrisJobke Yeah I think so, too - but how can I do this... Btw w
bildschirmfoto vom 2016-01-22 13 26 52
e have one more problem

The value of the input field will be reused

**Comment by rullzer** [comment]
I think I fixed the remaining issues here.

@Julian1998 @MorrisJobke @PVince81 please have a look.

**Comment by MorrisJobke** [comment]
Tested and works! Awesome :+1:

**Comment by LukasReschke** [comment]
Works :+1:

**Pull request merged**